erc/ERCS/erc-7562.md

```diff
@@ -317,19 +319,25 @@ Local storage rules protect the bundler against denial of service at the time of
   This means that `Paymaster` and `Factory` contracts cannot practically be an "account" contract as well.
 * **[STO-041]** `UserOperation` may not use associated storage (of either its account or from staked entity) in a contract that is a "sender" of another UserOperation in the mempool.
 
-### Staked Entities Reputation Rules
 
-* **[SREP-010]** The "canonical mempool" defines a staked entity if it has `MIN_STAKE_VALUE` and unstake delay of `MIN_UNSTAKE_DELAY`
-* **[SREP-020]** A `BANNED` address is not allowed into the mempool.\
+### General Reputation rules
+The following reputation rules apply for all staked entities, and for unstaked paymasters. All rules apply to all of these entities unless specified otherwise.
+
+* **[GREP-010]** A `BANNED` address is not allowed into the mempool.\
   Also, all existing `UserOperations` referencing this address are removed from the mempool.
-* **[SREP-030]** A `THROTTLED` address is limited to:
+* **[GREP-020]** A `THROTTLED` address is limited to:
     * `THROTTLED_ENTITY_MEMPOOL_COUNT` entries in the mempool.
     * `THROTTLED_ENTITY_BUNDLE_COUNT` `UserOperations` in a bundle.
     * Can remain in the mempool only for `THROTTLED_ENTITY_LIVE_BLOCKS`.
-* **[SREP-040]** An `OK` staked entity is unlimited by the reputation rule.
+* **[GREP-040]** If an entity fails the bundle creation after passing second validation, its `opsSeen` set to `BAN_OPS_SEEN_PENALTY`, and `opsIncluded` to zero, causing it to be `BANNED`.
+
+### Staked Entities Reputation Rules
+
+* **[SREP-010]** The "canonical mempool" defines a staked entity if it has `MIN_STAKE_VALUE` and unstake delay of `MIN_UNSTAKE_DELAY`
+* **[SREP-020]** An `OK` staked entity is unlimited by the reputation rule.
     * Allowed in unlimited numbers in the mempool.
     * Allowed in unlimited numbers in a bundle.
-* **[SREP-050]** If a staked entity fails the second validation or fails bundle creation, its `opsSeen` set to `10000`, and `opsIncluded` to zero, causing it to be `BANNED`.
+* **[SREP-050]** If a staked entity fails the second validation or fails bundle creation, its `opsSeen` is incremented by `BAN_OPS_SEEN_PENALTY`, causing it to be `BANNED`.
```

Comment:
```suggestion
* **[SREP-050]** If a staked entity fails the second validation or fails bundle creation, its `opsSeen` is set to `BAN_OPS_SEEN_PENALTY` and its `opsIncluded` is set to zero, causing it to be `BANNED`.
```

Why did this change to "increment opsSeen by X" after we already fixed it to "set opsSeen to X and opsIncluded to zero"? The behavior in this new version depends heavily on the current values, and may result in not banning for the expected duration (or at all if opsIncluded is sufficiently high).

Reply:
Damn!  It removed my 2nd suggestion.  Please do not accept until I re-add it

Reply:
```suggestion
```

Suggesting to remove this rule as it seems to be covered by GREP-040.  The rule also has the wrong condition but I won't correct it if we can just remove it.